File: scratch/recover_script.js
const fs = require('fs');

const transcriptPath = 'C:\\Users\\mafre\\.gemini\\antigravity-cli\\brain\\6ff176ef-c942-4d32-a127-da2d0fe58d52\\.system_generated\\logs\\transcript_full.jsonl';
const targetFile = 'BitacorasApp.tsx';
const lines = fs.readFileSync(transcriptPath, 'utf8').split('\n').filter(Boolean);

let content = '';

for (const line of lines) {
  try {
    const obj = JSON.parse(line);
    if (obj.type === 'PLANNER_RESPONSE' && obj.tool_calls) {
      for (const tc of obj.tool_calls) {
        if (tc.args && tc.args.TargetFile && tc.args.TargetFile.includes(targetFile)) {
          console.log(Applying step : );
          
          if (tc.name === 'write_to_file') {
            content = tc.args.CodeContent;
          } 
          else if (tc.name === 'replace_file_content') {
            const lines = content.split('\n');
            const start = tc.args.StartLine - 1;
            const end = tc.args.EndLine;
            // The TargetContent is expected to match precisely, but let's just replace the lines
            const replacement = tc.args.ReplacementContent.split('\n');
            content = [...lines.slice(0, start), ...replacement, ...lines.slice(end)].join('\n');
          }
          else if (tc.name === 'multi_replace_file_content') {
            // apply chunks in reverse order to not mess up line numbers
            const chunks = tc.args.ReplacementChunks.sort((a, b) => b.StartLine - a.StartLine);
            let lines = content.split('\n');
            for (const chunk of chunks) {
              const start = chunk.StartLine - 1;
              const end = chunk.EndLine;
              const replacement = chunk.ReplacementContent.split('\n');
              lines = [...lines.slice(0, start), ...replacement, ...lines.slice(end)];
            }
            content = lines.join('\n');
          }
        }
      }
    }
  } catch (e) {
    // ignore parse errors
  }
}

fs.writeFileSync('C:\\Users\\mafre\\Esolenergias\\scratch\\RecoveredBitacorasApp.tsx', content, 'utf8');
console.log('Recovery complete! Length:', content.split('\n').length);

File: scratch/recover_script.py
import json
import os

transcript_path = r"C:\Users\mafre\.gemini\antigravity-cli\brain\6ff176ef-c942-4d32-a127-da2d0fe58d52\.system_generated\logs\transcript_full.jsonl"
target_file = "BitacorasApp.tsx"

content = ""

with open(transcript_path, "r", encoding="utf-8") as f:
    for line in f:
        if not line.strip(): continue
        try:
            obj = json.loads(line)
            if obj.get("type") == "PLANNER_RESPONSE" and "tool_calls" in obj:
                for tc in obj["tool_calls"]:
                    args = tc.get("args", {})
                    if args.get("TargetFile", "").endswith(target_file):
                        name = tc.get("name")
                        print(f"Applying {name} at step {obj['step_index']}")
                        
                        if name in ["write_to_file", "write_file"]:
                            content = args.get("CodeContent", "")
                        
                        elif name == "replace_file_content":
                            target = args.get("TargetContent", "")
                            replacement = args.get("ReplacementContent", "")
                            if target in content:
                                content = content.replace(target, replacement, 1)
                            else:
                                print(f"WARNING: Target not found in step {obj['step_index']}")
                        
                        elif name == "multi_replace_file_content":
                            chunks = args.get("ReplacementChunks", [])
                            for chunk in chunks:
                                target = chunk.get("TargetContent", "")
                                replacement = chunk.get("ReplacementContent", "")
                                if target in content:
                                    content = content.replace(target, replacement, 1)
                                else:
                                    print(f"WARNING: Chunk target not found in step {obj['step_index']}")
        except Exception as e:
            pass

with open(r"C:\Users\mafre\Esolenergias\scratch\RecoveredBitacorasApp_v2.tsx", "w", encoding="utf-8") as f:
    f.write(content)

print(f"Done! Recovered length: {len(content.splitlines())}")
